shared/ssz/decode.go

```diff
@@ -216,6 +273,31 @@ func makeStructDecoder(typ reflect.Type) (decoder, error) {
 	return decoder, nil
 }
 
+// Notice: Currently we don't support nil pointer:
+// - Input for encoding must not contain nil pointer
```

Comment:
Is it possible to through an informative error if a user passes in a nil pointer?

Reply:
We couldn't do that for decode target. When passing a decode target object, it could be uninitialized, so it can have nil pointer in its fields. As we decode we'll fill the nil pointer with the actual content. So no matter what pointer value you have in the target (nil or not), it will be filled with value decoded from the input stream.